src/generic/treectlg.cpp

```diff
@@ -3010,6 +3010,10 @@ void wxGenericTreeCtrl::OnSize( wxSizeEvent &event )
 void wxGenericTreeCtrl::OnPaint( wxPaintEvent &WXUNUSED(event) )
 {
     wxPaintDC dc(this);
+    #ifdef __WXOSX__
+    // at least on OSX otherwise the non-painted area remains transparent
```

Comment:
This would seem to indicate that wxOSX doesn't implement the contract of wx API correctly: it's supposed to erase background with the default `wxBG_STYLE_ERASE` background style, so that `wxEVT_PAINT` handler could paint only the "foreground" over it.

Note that in this particular case we actually could set the background style to `wxBG_STYLE_PAINT` and call `dc.Clear()` here too, but the current code should still work, so it looks like something could be wrong in `wxWindowMac::MacDoRedraw()`, could you please check it?